fix: reset Cloudflare Turnstile token on signup errors

dhairyashiil opened the pull request:
### Invalid Cloudflare token (Sign-up)
Steps to reproduce the issue:

Try this:
1. Go to [sign-up](https://app.cal.com/signup)
2. Random user name
3. Use an already taken email, for example [carina@cal.com](mailto:carina@cal.com)
4. Click ‘Create account’
5. Expected error ‘Username or email is already taken’
6. Change username
7. Click ‘Create Account’
8. ‘Invalid cloudflare token’ error

<img width="591" height="390" alt="Screenshot 2025-07-07 at 11 02 26 AM" src="https://github.com/user-attachments/assets/06be94aa-1f7f-4ab0-b932-3ca24c344285" />

Fixes #25762
Fixes CAL-6896


### Before: 



https://github.com/user-attachments/assets/f18d241c-8ec1-463e-a118-0cac884c3bc2




### After:



https://github.com/user-attachments/assets/14f74cbd-702f-44cc-a309-b68bcea917b7



## What does this PR do?

### The Problem:

When users tried to sign up with an email that was already taken, they would see an error message. However, if they then changed their username and tried again, they would get a confusing "Invalid cloudflare token" error instead of being able to complete signup. This forced users to refresh the page and start over.

### What Was Wrong:

Our signup form uses Cloudflare Turnstile (a security verification widget) to prevent bots. Each verification generates a one-time-use token that gets sent to the server. The issue was that after the first failed signup attempt, the form kept reusing the same expired token instead of generating a fresh one. When the server received the already-used token on the second attempt, it rejected it as invalid.

### The Fix:

We now automatically reset the Turnstile widget after any signup error. This generates a new verification token, allowing users to retry their signup without refreshing the page. We also added handlers to clear expired tokens and prevent submission when no valid token is available.

### Impact:

Users can now fix validation errors (like using a taken email or username) and retry signup immediately without encountering the "Invalid cloudflare token" error or needing to refresh the page.

dhairyashiil committed:
fix: cloudflare turnstile token reset

Udit-takkar commented on apps/web/modules/signup-view.tsx:
```diff
@@ -209,11 +211,17 @@ export default function Signup({
+    if (CLOUDFLARE_SITE_ID) {
+      console.log("Turnstile ref available:", Boolean(turnstileRef.current));
+    }
+  }, []);
```
Do we want to log this on prod?

Udit-takkar commented on apps/web/modules/signup-view.tsx:
```diff
@@ -299,7 +318,19 @@ export default function Signup({
-        formMethods.setError("apiError", { message: err.message });
+        const message =
+          err.message === "Invalid cloudflare token"
+            ? "Verification expired. Please try again."
```
nit

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into fix/cloudflare-turnstile-token-reset

dhairyashiil committed:
Merge branch 'main' into fix/cloudflare-turnstile-token-reset

dhairyashiil committed:
fix: silently reset turnstile on invalid token error

dhairyashiil committed:
Merge upstream/main and resolve conflicts

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into fix/cloudflare-turnstile-token-reset

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into fix/cloudflare-turnstile-token-reset

dhairyashiil committed:
refactor: remove unused forwardRef logic from Turnstile component

- Remove forwardRef, useImperativeHandle, and useRef imports
- Remove unused TurnstileInstance type export
- Simplify to a plain function component
- The ref-based reset was replaced by key-based remount in signup-view

dhairyashiil committed:
refactor: remove redundant cfToken validation check

The submit button is already disabled when cfToken is missing,
making this defensive check unreachable during normal form flow.

dhairyashiil committed:
revert prettier formatiing

dhairyashiil committed:
chore: revert yarn.lock changes

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into fix/cloudflare-turnstile-token-reset

Udit-takkar commented on apps/web/modules/signup-view.tsx:
```diff
@@ -306,6 +307,13 @@ export default function Signup({
+        formMethods.setValue("cfToken", undefined);
+
+        if (err.message === "Invalid cloudflare token") {
+          return;
```
If we change the error message in the future. we might break this

dhairyashiil committed:
refactor(auth): use shared constant for cloudflare token error message

Replace hardcoded "Invalid cloudflare token" string with an exported
constant to prevent silent breakage if the error message changes.

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into fix/cloudflare-turnstile-token-reset
